client/extensions/woocommerce/index.js

```diff
@@ -14,39 +14,84 @@ import ProductCreate from './app/products/product-create';
-		page( '/store/:site', siteSelection, navigation, Controller.dashboard );
-		page( '/store/products/:site/add', siteSelection, navigation, Controller.addProduct );
-	}
+	const storePages = [
+		{
+			container: Dashboard,
+			configKey: 'woocommerce/extension-dashboard',
+			route: '/store/:site',
+		},
+		{
+			container: Dashboard, // TODO use Dashboard as a placeholder until this page becomes available
+			configKey: 'woocommerce/extension-products',
+			route: '/store/products/:site',
+		},
+		{
+			container: ProductCreate,
+			configKey: 'woocommerce/extension-products',
+			route: '/store/products/:site/add',
+		},
+		{
+			container: Dashboard, // TODO use Dashboard as a placeholder until this page becomes available
+			configKey: 'woocommerce/extension-products-import',
+			route: '/store/products/:site/import',
+		},
+		{
+			container: Dashboard, // TODO use Dashboard as a placeholder until this page becomes available
+			configKey: 'woocommerce/extension-orders',
+			route: '/store/orders/:site',
+		},
+		{
+			container: Dashboard, // TODO use Dashboard as a placeholder until this page becomes available
+			configKey: 'woocommerce/extension-orders',
+			route: '/store/orders/:site/add',
+		},
+		{
+			container: Dashboard, // TODO use Dashboard as a placeholder until this page becomes available
+			configKey: 'woocommerce/extension-promotions',
+			route: '/store/promotions/:site',
+		},
+		{
+			container: Stats,
+			configKey: 'woocommerce/extension-stats',
+			route: '/store/stats/:site',
+		},
+		{
+			container: Dashboard, // TODO use Dashboard as a placeholder until this page becomes available
+			configKey: 'woocommerce/extension-extensions',
+			route: '/store/extensions/:site',
+		},
+		{
+			container: Dashboard, // TODO use Dashboard as a placeholder until this page becomes available
+			configKey: 'woocommerce/extension-settings',
+			route: '/store/settings/:site',
+		},
+		{
+			container: Dashboard, // TODO use Dashboard as a placeholder until this page becomes available
+			configKey: 'woocommerce/extension-settings-checkout',
+			route: '/store/settings/:site/checkout',
+		},
+		{
+			container: Dashboard, // TODO use Dashboard as a placeholder until this page becomes available
+			configKey: 'woocommerce/extension-settings-shipping',
+			route: '/store/settings/:site/shipping',
+		},
+		{
+			container: Dashboard, // TODO use Dashboard as a placeholder until this page becomes available
+			configKey: 'woocommerce/extension-settings-tax',
+			route: '/store/settings/:site/tax',
+		},
+	];
 
-	if ( config.isEnabled( 'woocommerce/extension-stats' ) ) {
-		page( '/store/stats/:site', siteSelection, navigation, Controller.stats );
-	}
+	storePages.forEach( function( storePage ) {
+		if ( config.isEnabled( storePage.configKey ) ) {
+			page( storePage.route, siteSelection, navigation, function( context ) {
+				renderWithReduxStore(
+					React.createElement( storePage.container, { } ),
```

Comment:
How about we pass `woocommerce` in as a className prop instead of adding nesting? Or are you foreseeing other needs for the wrapper div?